README.md

```diff
@@ -0,0 +1,180 @@
+
+### Tags in A3
+
+In A2, there is always an `apostrophe-tags` module to manage tags, and every piece has an array of tag names in its `tags` property. These all come from the same namespace.
+
+In A3 there is no such type. Instead, you can create new piece types as tags for other piece types and use `relationship` fields to connect your pieces to them. This avoids an explosion of poorly-curated tags and keeps them relevant to the right kind of content.
+
+Many A2 sites don't really use tags. However if you do, keep in mind that the `tags` property remains in the database after the upgrade. Although you could use a global transformation function to convert these to a relationship (see below), it may be easier to do that with a migration in your A3 code after the initial transition to A3.
+
+### Global transformation functions
+
+Most of the time it makes sense to write a transformation function for each doc or widget type. But if you need to address something for every doc and widget, you can write a global transformation function:
+
+```javascript
+module.exports = {
+  async transformDoc(doc) {
+    // This function is invoked for every doc type
+    return {
+      ...doc,
+      // changed properties here
+    };
+  },
+  async transformWidget(widget) {
+    // This widget is invoked for every widget type
+    return {
+      ...widget,
+      // changed properties here
+    };
+  }
+}
+```
+
+Note that as before, transformation functions **must** return a doc or widget, as appropriate. You can modify the original but you must return it.
+
+> Not all transformations are easiest to achieve during the upgrade. Some might be more easily achieved in A3 after the initial upgrade. Keep in mind that even if a property is not part of the A3 schema, it will remain in the database.
+
+## Running the upgrade
+
+You'll need a new A3 project to copy the content to. Specify both the new project folder and the MongoDB URI of the new A3 database. If you are running MongoDB locally, which is typical during development, you can specify:
+
+```
+mongodb://localhost:27017/your-new-database-name
+```
+
+> 🛑 Don't use the same name as your existing A2 database. Your new A3 project should also have a different `shortName` setting in `app.js`, for avoidance of any possible confusion. Also make sure your A3 project is not already running, for instance in another terminal window.
+
+```
+node app @apostrophecms/content-upgrader:upgrade --a3-project=../your-new-a3-project-folder --a3-db=mongodb://localhost:27017/your-new-database-name
+```
+
+## Options
+
+### `--skip-uploads`
+
+By default this tool will also copy media from `public/uploads` to the `public/uploads` folder of the A3 project, using `rsync`. You can skip this step with `--skip-uploads`. Note that **the actual contents of the folder won't change**, so feel free to copy it by other means, especially if you are using AWS S3 for media storage, etc.
+
+## Next steps
```

Comment:
Should the tool end with a list of any things we think they might need to manually change? Like the image widget with multiple items.

Reply:
That would be good. PRO-2095